src/main/java/frc/robot/filters/AutoYawFilter.java

```diff
@@ -0,0 +1,51 @@
+package frc.robot.filters;
+
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.wpilibj.interfaces.Gyro;
+
+/**
+ *
+ * 
+ * @author aheitkamp
+ */
+public class AutoYawFilter extends Filter {
+    private Gyro navX;
+    private PIDController PID;
+
+    public AutoYawFilter(Gyro NavX) {
+        navX = NavX;
+
+        PID = new PIDController(0.095, 0.15, 0.0085);
+        PID.setTolerance(2);
+        PID.enableContinuousInput(-180, 180);
+    }
+
+    public void filter(DriveInput DI) {
```

Comment:
i really dont like filtering by side affect like this. 
I'd prefer that filter return a new DriveInput, and the original to be unmodified. 
Also, DI should be di.